libpages/root.go

```diff
@@ -92,25 +107,28 @@ func (r *Root) MakeFS(
 		libkbfs.CtxWithRandomIDReplayable(
 			fsCtx, ctxIDKey, ctxOpID, nil))
 	if err != nil {
-		return nil, tlf.ID{}, nil, err
+		return CacheableFS{}, tlf.ID{}, nil, err
 	}
 	switch r.Type {
 	case KBFSRoot:
 		tlfHandle, err := libkbfs.GetHandleFromFolderNameAndType(
 			ctx, kbfsConfig.KBPKI(), kbfsConfig.MDOps(), r.TlfNameUnparsed, r.TlfType)
 		if err != nil {
-			return nil, tlf.ID{}, nil, err
+			return CacheableFS{}, tlf.ID{}, nil, err
 		}
-		fs, err = libfs.NewFS(fsCtx, kbfsConfig,
-			tlfHandle, r.PathUnparsed, "", keybase1.MDPriorityNormal)
+		tlfFS, err := libfs.NewFS(fsCtx, kbfsConfig,
+			tlfHandle, "", "", keybase1.MDPriorityNormal)
 		if err != nil {
-			return nil, tlf.ID{}, nil, err
+			return CacheableFS{}, tlf.ID{}, nil, err
 		}
-		return fs, tlfHandle.TlfID(), cancel, nil
+		return CacheableFS{
```

Comment:
Is it worth trying to do a `Use()` here, just to make sure `r.PathUnparsed` is valid before we return (and presumably cache) the object?